### ISSUE TITLE
wkhtmltoimage: load-error-handling and load-media-error-handling  are ignored.

### ISSUE THREAD

**Issue by martinnj** [wkhtmltopdf/wkhtmltopdf#4525]
**wkhtmltopdf version(s) affected**: 0.12.5

**OS information**
- `Ubuntu Bionic 18.0.4 LTS` (package installed manually from `https://downloads.wkhtmltopdf.org/0.12/0.12.5/wkhtmltox_0.12.5-1.bionic_amd64.deb`)
- `4.19.79-1-MANJARO` (package installed from Community repository)

**Description**
When rendering a webpage or html document to an image, the error handling command line options are ignored.
Specifically:
` --load-error-handling ignore --load-media-error-handling ignore `.
Instead it appears to treath both as "abort".

**How to reproduce**
Save the following to `input.html`:
```
<img src="https://www.abbuildingproducts.co.uk/media/catalog/product" />
```
The URL above returns 403 Forbidden, it is an invalid URL on purpose. :)

Then run:
```
wkhtmltoimage --format png --load-error-handling ignore --load-media-error-handling ignore input.html image.png
```

**Expected behavior**
A file in `image.png` with the `<img>` element missing., so a white image, depending on X/Qt settings. And maybe some log messages about the URL in the element being ignored/skipped.

**Possible Solution**  
The options from `wkhtmltoimage --help` doesn't list the options specifically, but they do get accepted (invalid options usually result in an error and the help being printed).
I can also see that the options are supported as "common" though: https://github.com/wkhtmltopdf/wkhtmltopdf/blob/master/src/shared/commonarguments.cc#L213-L214
Wether they do anything or not I haven't checked, but they would be very useful to have working.

If you need anything else, please let me know. :)

**Comment by jonathanbull** [comment]
Bumping this one as we're also having the same issue :)

**Comment by ABHISHEK-KEDAR-21** [comment]
Getting this error:
wkhtmltoimage exited with non-zero code -9. error:
Rendering (2/2)                                                    
[>                                                           ] 0%
[===============>                                            ] 25%

**Comment by alessionossa** [comment]
Same problem here with wkhtmltopdf 0.12.6

**Comment by T35R6braPwgDJKq** [comment]
same here, any workaround?

**Comment by martinnj** [comment]
> 
> 
> same here, any workaround?

Yes, but it becomes really jank really fast:
- You can check for the existence of the output file.
- There is also some parsing options if you output to console instead. You can inspect the bytes for the header of the image format you have requested.

**Comment by T35R6braPwgDJKq** [comment]
> You can check for the existence of the output file.

You mean on another error the output file would not be generated?

**Comment by lengyuan** [comment]
getting same issue, have tried
--load-error-handling ignore --load-media-error-handling ignore
--load-error-handling skip --load-media-error-handling skip
but none of these works

**Comment by tod222** [comment]
Look at [multipageloader.cc](https://github.com/wkhtmltopdf/wkhtmltopdf/blob/5b26acc41d8209d3b4c4018ac7350b8cea64fa73/src/lib/multipageloader.cc). ResourceObject::amfinished exits with the error before checking load-media-error-handling.

Affects wkhtmltopdf also, where I'm seeing it.

**Comment by imuller** [comment]
Same here, not working when using --load-media-error-handling

**Comment by imuller** [comment]
You need to set the argument at the end of the command. For example:

`wkhtmltopdf test.html test.pdf --load-media-error-handling abort`

**Comment by cherchi-ow** [comment]
Same problem, did you find any workaround?

**Comment by martinnj** [comment]
> 
> 
> Same problem, did you find any workaround?

Yeah, check https://github.com/wkhtmltopdf/wkhtmltopdf/issues/4525#issuecomment-781988755
